### PR TITLE
JAMES-2589 [Swift BlobStore] Module should support PayloadCodec configuration

### PR DESCRIPTION
**Pull request by trantienduchn**

### CODE REVIEW

File: server/container/guice/blob-objectstorage-guice/src/main/java/org/apache/james/modules/objectstorage/ObjectStorageDependenciesModule.java [hunk 1]
```diff
@@ -96,4 +91,7 @@ private ObjectStorageBlobsDAO buildObjectStore(ObjectStorageBlobConfiguration co
         }
     }
 
+    private PayloadCodec buildPayloadCodec(ObjectStorageBlobConfiguration configuration) {
+        return configuration.getPayloadCodecFactory().create(configuration);
```
Comment (chibenwa) [inline server/container/guice/blob-objectstorage-guice/src/main/java/org/apache/james/modules/objectstorage/ObjectStorageDependenciesModule.java]: This line seems boiler plate.

Can't we have a simple `configuration.getPayloadCodec()` ?

**Comment by mbaechler** [comment]
Of course an integration test would be welcome

**Comment by chibenwa** [comment]
test this please

File: server/blob/blob-objectstorage/src/main/java/org/apache/james/blob/objectstorage/ObjectStorageBlobsDAO.java [hunk 2]
```diff
@@ -147,4 +147,9 @@ public InputStream read(BlobId blobId) throws ObjectStoreException {
     public void deleteContainer() {
         blobStore.deleteContainer(containerName.value());
     }
+
+    @VisibleForTesting
+    public PayloadCodec getPayloadCodec() {
```
Comment (aduprat) [inline server/blob/blob-objectstorage/src/main/java/org/apache/james/blob/objectstorage/ObjectStorageBlobsDAO.java]: You don't need `public` when have the `VisibleForTesting` annotation.

**Comment by chibenwa** [comment]
test this please

File: server/container/guice/cassandra-rabbitmq-guice/src/test/java/org/apache/james/CassandraRabbitMQJamesServerTest.java [hunk 3]
```diff
@@ -87,7 +93,29 @@ default void mailsShouldBeWellReceived(GuiceJamesServer server) throws Exception
 
     @Nested
     @TestInstance(Lifecycle.PER_CLASS)
-    class WithSwift implements ContratSuite {
+    class WithEncryptedSwift implements ContractSuite {
+        @RegisterExtension
+        JamesServerExtension testExtension = new JamesServerExtensionBuilder()
```
Comment (hoangdat) [inline server/container/guice/cassandra-rabbitmq-guice/src/test/java/org/apache/james/CassandraRabbitMQJamesServerTest.java]: We duplicate this

**Comment by chibenwa** [comment]
Merged

**Pull request closed without merging**